src/github/github-api.spec.js

```diff
@@ -129,19 +136,25 @@ describe("repository operations", function () {
       expect(data.content.sha).toBeDefined();
       sha = data.content.sha;
       done()
+    }).catch(error => {
+      console.log(error);
+      fail()
     });
   });
 
   it("delete file in repository", function(done){
     jasmine.DEFAULT_TIMEOUT_INTERVAL= 2000;
-    gh.deleteFile('episodeyang', 'eywa-github',
+    console.log(sha)
+    gh.deleteFile(username, testRepoName,
       '/test_folder/test_file.md',
       "test commit from eywa-github driver, delete file",
       sha
     ).then(data => {
       done()
+    }).catch(error => {
+      console.log(error);
+      fail()
```

Comment:
you should
1. use `throw Error` in this case
2. only call the `done` callback in final (not sure if this is correct b/c forgot the logic of `fetch api`.